internal/controllers/topology/cluster/cluster_controller.go

```diff
@@ -200,6 +208,17 @@ func (r *Reconciler) reconcile(ctx context.Context, s *scope.Scope) (ctrl.Result
 		return ctrl.Result{}, errors.Wrap(err, "error reading current state of the Cluster topology")
 	}
 
+	// The cluster topology is yet to be created. Call the BeforeClusterCreate hook before proceeding.
+	if feature.Gates.Enabled(feature.RuntimeSDK) {
+		res, err := r.callBeforeClusterCreateHook(ctx, s)
+		if err != nil {
+			return reconcile.Result{}, errors.Wrap(err, "error calling BeforeClusterCreate hook")
```

Comment:
```suggestion
			return reconcile.Result{}, err
```
nit: we are already building this error internally